bundles/org.openhab.ui/web/src/assets/definitions/blockly/ohblocks_ephemeris.js

```diff
@@ -0,0 +1,128 @@
+/*
+* Ephemeris provides calendar related information
+* This code has been originally provided by https://github.com/bigbasec
+* @author stefan.hoehn
+*
+* See more background info on openHAB ephemeris here: https://www.openhab.org/docs/configuration/actions.html#ephemeris
+*/
+import Blockly from 'blockly'
+import { FieldItemModelPicker } from './ohitemfield'
+
+export default function defineOHBlocks_Ephemeris (f7) {
+  Blockly.Blocks['oh_ephemeris_basic'] = {
+    init: function () {
+      this.appendDummyInput()
+        .appendField(new Blockly.FieldDropdown([['isWeekend', 'isWeekend'], ['getBankHolidayName', 'getBankHolidayName'], ['getNextBankHoliday', 'getNextBankHoliday'], ['isBankHoliday', 'isBankHoliday']]), 'type')
+      this.setOutput(true, null)
+      this.setColour(230)
+      this.setTooltip('provides calendar related information')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/actions.html#ephemeris')
+    }
+  }
+
+  Blockly.JavaScript['oh_ephemeris_basic'] = function (block) {
+    addEphemeris()
+    let type = block.getFieldValue('type')
+    let code = `ephemeris.${type}()`
+    return [code, Blockly.JavaScript.ORDER_NONE]
+  }
+
+  Blockly.Blocks['oh_ephemeris_offset'] = {
+    init: function () {
+      this.appendValueInput('offset')
+        .setCheck('Number')
+        .appendField(new Blockly.FieldDropdown([['isWeekend', 'isWeekend'], ['getBankHolidayName', 'getBankHolidayName'], ['getNextBankHoliday', 'getNextBankHoliday'], ['isBankHoliday', 'isBankHoliday']]), 'type')
+        .appendField('Offset Days')
+      this.setOutput(true, null)
+      this.setColour(230)
+      this.setTooltip('the offset to the given type in days')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/actions.html#ephemeris')
+    }
+  }
+
+  Blockly.JavaScript['oh_ephemeris_offset'] = function (block) {
+    addEphemeris()
+    let type = block.getFieldValue('type')
+    let offsetValue = Blockly.JavaScript.valueToCode(block, 'offset', Blockly.JavaScript.ORDER_ATOMIC)
+    let code = `ephemeris.${type}(${offsetValue})`
+    return [code, Blockly.JavaScript.ORDER_NONE]
+  }
+
+  Blockly.Blocks['oh_Ephemeris_getBankHolidayName'] = {
```

Comment:
This doesn't appear in the toolbox.
Also please use consistent casing (lowercase snake_case).

Reply:
good catch - these would have been the individual ones that you had asked for above only that haven't been put into the editor file (but without the offset support). As mentioned above if we did it this way, we would end up in 8 different blocks.